app/Models/User.php

```diff
@@ -609,4 +607,50 @@ public function scopeTracked(Builder $query): Builder
     {
         return $query->where('Untracked', false); // TODO: use unranked_at=NULL?
     }
+
+    /**
+     * @param Builder<User> $query
+     * @return Builder<User>
+     */
+    public function scopeSearch(Builder $query, string $keyword): Builder
+    {
+        // Just return the base query if the keyword is empty.
+        if (empty($keyword)) {
+            return $query->whereNotNull('email_verified_at')
+                ->whereNull('Deleted')
+                ->whereNull('banned_at');
+        }
+
+        $keyword = trim($keyword);
+        $escapedKeyword = addslashes($keyword);
+        $fullTextQuery = '+' . $escapedKeyword . '*';
+
+        /**
+         * Search users with FULLTEXT index and rank results by:
+         * 1. Exact username match (100 points).
+         * 2. Exact display name match (90 points).
+         * 3. Username prefix match (80 points).
+         * 4. Display name prefix match (70 points).
+         * 5. Username infix match (weighted by FULLTEXT relevance).
+         * 6. Display name infix match (weighted by FULLTEXT relevance).
```

Comment:
As User is not supposed to be visible for users with a custom display_name, display_name should be prioritized over User and "User like" should be eliminated in favor of just letting the match handle it. (display_name=100, User=90, like display_name%=80, match result)

Reply:
Tweaked this in latest - here's the new scoring:
```php
/**
 * Search users and rank results by:
 * 1. Exact display_name match (100 points).
 * 2. display_name prefix match (90 points).
 * 3. display_name contains match (80 points).
 * 4. Exact username match (70 points).
 * 5. username contains match (60 points).
 *
 * Finally, sort equally-ranked results by last login date.
 */
```
#5 (`User LIKE`) is there as a result of the removal of the FULLTEXT index.